### ISSUE TITLE
How do I get the Windows 8.1 version

### ISSUE THREAD

**Issue by mediabuff** [Microsoft/cppwinrt#250]

**Comment by Scottj1s** [comment]
The challenge here is that while C++/WinRT targets the latest SDK (Creator's Update, 15063), which is compatible with Windows 8.1, it also takes advantage of a number of C++ compiler features that are only available with Visual Studio 2017.  And Visual Studio 2017 no longer supports building Windows 8.1 Phone or Desktop apps.  For that, you have to fall back to Visual Studio 2015, installing the optional "Windows 8.1 and Windows Phone 8.0/8.1" support.  VS 2015 can be installed side by side with VS 2017.  You could then create a universal Windows Runtime component with C++/WinRT in VS 2017, and bundle it with a Windows 8.1 app built in VS 2015.  Not ideal, and Kenny may have additional suggestions.

**Comment by kennykerr** [comment]
It depends on what you're trying to achieve. Visual Studio 2017 [doesn't appear to support building store apps](https://www.visualstudio.com/en-us/productinfo/vs2017-compatibility-vs) for Windows 8.1 but it does support desktop/console apps and you should still be able to use C++/WinRT in that environment.

**Comment by mediabuff** [comment]
Thanks for all your guidance. This repo used to have the Win81 version - how do I get hold of it ? The current version doesn't compile. Yes, I am using VS 2015 to continue support my Win81 apps, but steadily dual platforming the shared modules. Ability to use cppwinrt is essential for that.

**Comment by kennykerr** [comment]
In that case you simply need to [git fetch the appropriate commit](https://github.com/Microsoft/cppwinrt/commits/master).

**Comment by mediabuff** [comment]
I checked all the old commits. The 8.1 folder seems to have disappeared. The root folder had a sub folder for 8.1

**Comment by Scottj1s** [comment]
I'm not aware of an 8.1 folder specifically.  I think all projections have been based on SDK version.  If you require the older headers or support for Visual Studio 2015, you can sync to a previous commit such as the [February 2017 Release](https://github.com/Microsoft/cppwinrt/tree/february_2017_release).

**Comment by walbourn** [comment]
To my knowledge there's never been a version of the C++/WinRT projection headers that was produced for the Windows 8.1 SDK. The first 'official' release was for use with the Windows 10 SDK (14393) and VS 2015 which is [here](https://github.com/Microsoft/cppwinrt/releases/tag/april_2017_release). To use the latest version of the C++/WinRT projection headers you must be using the Windows SDK (15063) which requires VS 2017 (ideally the 15.3 update).

To build Windows 8.1 Store apps, you need to use the Windows 8.1 SDK and the VS 2013 toolset (``v120``). VS 2017 doesn't support this configuration, but VS 2015 does--there's an option in the Custom Install to add the Windows 8.1 tools. You can build Win32 desktop apps with VS 2015 or VS 2017 (with the appropriate workload installed).

**So in short, the original question doesn't make a lot of sense. What are you trying to build?**

**Comment by mediabuff** [comment]
Thank you. My lapse of memory :-)
Yes, I understand and I use VS2015 to support and maintain Win81 app. Also, have a Win10 version with lot of shared components.
The original request is simple - how do I have develop Win81 modules using cppwinrt. It's understood that we leverage only platform API's that are available in Win81.

**Comment by mediabuff** [comment]
I finally figured. These are located in https://github.com/kennykerr/modern
